src/base/bittorrent/session.cpp

```diff
@@ -531,6 +481,11 @@ Session::Session(QObject *parent)
     m_resumeDataSavingManager = new ResumeDataSavingManager(m_resumeFolderPath);
     m_resumeDataSavingManager->moveToThread(m_ioThread);
     connect(m_ioThread, &QThread::finished, m_resumeDataSavingManager, &QObject::deleteLater);
+
+    m_resumeDataLoadingManager = new ResumeDataLoadingManager(m_resumeFolderPath);
+    m_resumeDataLoadingManager->moveToThread(m_ioThread);
+    connect(m_ioThread, &QThread::finished, m_resumeDataLoadingManager, &QObject::deleteLater);
```

Comment:
You need also add:
```c++
connect(m_resumeDataLoadingManager, &ResumeDataLoadingManager::loadingFinished, m_resumeDataLoadingManager, &QObject::deleteLater);
```

Also I would move its initialization it `startUpTorrents()` and make it local variable since it isn't needed anywhere else.